lib.js

```diff
@@ -0,0 +1,215 @@
+// TODO
+// - decide where to update window.localstorage.{lastCid,ipns,lastPublishedCid}
+// - make function names consistent nounVerb or verbNoun
+
+
+export async function publish(cid){
+  let r = await fetch('ipns://ipmb', {
+    method: 'POST', 
+    body: cid,
+  });
+  const ipns = await r.text();
+  const updateHtml = !window.localStorage.ipns;
+  window.localStorage.ipns = ipns;
+  window.localStorage.lastPublishedCid = cid;
+  return ipns;
+}
+
+export function parseFilename(filename){
+  let date = filename.slice(0, 10);
+  let title = decodeURIComponent(filename.slice(11)).replace(/\.md$/, '');
+  return { date, title };
+}
+
+export function filenameToTitle(filename){
+  return decodeURIComponent(filename.slice(11)).replace(/\.md$/, '');
+}
+
+function parseMeta(file){
+  return {
+    ...file,
+    title: filenameToTitle(file.filename),
+    date: file.filename.slice(0, 10),
+    excerpt: file.content.slice(0, Math.min(file.content.indexOf('\n'), 100)),
+  }
+}
+
+
+// Create markdown blog based on data stored in lastCid
+export async function thisIsBlog(doFileStuff){
+
+  let lastCid = window.localStorage.getItem('lastCid');
+  const previousCid = lastCid;
+ 
+  // make the update
+  lastCid = await doFileStuff();
```

Comment:
This could be refactored. The CID used by `doFileStuff` was passed to it in a closure, based on the value in localStorage. That works since it's the first operation being done in the `thisIsBlog` function